src/v0/destinations/marketo_static_list/transform.js

```diff
@@ -71,76 +72,94 @@ const processEvent = (input) => {
-    throw new InstrumentationError(
-      'Invalid leadIds format or no leadIds found neither to add nor to remove',
-    );
   }
   return response;
 };
-const process = async (event) => {
-  const token = await getAuthToken(formatConfig(event.destination));
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const process = async (event, _processParams) => {
+  const token = await getAuthToken(formatConfig(event.destination));
   if (!token) {
     throw new UnauthorizedError('Authorization failed');
   }
-  const response = processEvent({ ...event, token });
+  const updatedEvent = { ...event, token };
+  const response = processEvent(updatedEvent);
   return response;
 };
 
 const processRouterDest = async (inputs, reqMetadata) => {
   // Token needs to be generated for marketo which will be done on input level.
   // If destination information is not present Error should be thrown
-  let token;
+
+  const { destination } = inputs[0];
   try {
-    token = await getAuthToken(formatConfig(inputs[0].destination));
+    const token = await getAuthToken(formatConfig(destination));
     if (!token) {
-      const errResp = {
-        status: 400,
-        message: 'Authorisation failed',
-        responseTransformFailure: true,
-        statTags: {},
-      };
-      const respEvents = getErrorRespEvents(
-        inputs.map((input) => input.metadata),
-        errResp.status,
-        errResp.message,
-        errResp.statTags,
-      );
-      return [{ ...respEvents, destination: inputs?.[0]?.destination }];
+      throw new UnauthorizedError('Could not retrieve authorisation token');
     }
   } catch (error) {
-    // Not using handleRtTfSingleEventError here as this is for multiple events
-    const errObj = generateErrorObject(error);
-    const respEvents = getErrorRespEvents(
-      inputs.map((input) => input.metadata),
-      errObj.status,
-      errObj.message,
-      errObj.statTags,
+    const errorObj = generateErrorObject(error);
+    const errResponses = inputs.map((input) =>
+      getErrorRespEvents(input.metadata, errorObj.status, errorObj.message, errorObj.statTags),
     );
-    return [{ ...respEvents, destination: inputs?.[0]?.destination }];
+
+    return errResponses;
   }
 
-  // Checking previous status Code. Initially setting to false.
-  // If true then previous status is 500 and every subsequent event output should be
-  // sent with status code 500 to the router to be retried.
-  const tokenisedInputs = inputs.map((input) => ({ ...input, token }));
-  const respList = await simpleProcessRouterDest(tokenisedInputs, processEvent, reqMetadata);
+  // use lodash.groupby to group the inputs based on message type
+  const transformedRecordEvent = [];
+  let transformedAudienceEvent = [];
+  const groupedInputs = lodash.groupBy(inputs, (input) => input.message.type);
+
+  const respList = [];
+  // process record events
+  if (Array.isArray(groupedInputs.record) && groupedInputs.record.length > 0) {
```

Comment:
empty group is not possible right? can we simplify the condition to `if(groupedInputs.record) {`

Reply:
Empty group is possible for cases when only audienceList events are sent through.